raiden-ts/src/services/epics.ts

```diff
@@ -320,43 +323,50 @@ export function pfsFeeUpdateEpic(
 ): Observable<messageGlobalSend.request> {
   return state$.pipe(
     groupChannel$,
-    // get only first state per channel
-    mergeMap((grouped$) => grouped$.pipe(first())),
-    withLatestFrom(config$),
-    // ignore actions while/if mediating not enabled
-    filter(
-      ([channel, { pfsRoom, caps }]) =>
-        channel.state === ChannelState.open && !!pfsRoom && !!getCap(caps, Capabilities.MEDIATE),
-    ),
-    mergeMap(([channel, { pfsRoom }]) => {
-      const message: PFSFeeUpdate = {
-        type: MessageType.PFS_FEE_UPDATE,
-        canonical_identifier: {
-          chain_identifier: BigNumber.from(network.chainId) as UInt<32>,
-          token_network_address: channel.tokenNetwork,
-          channel_identifier: BigNumber.from(channel.id) as UInt<32>,
-        },
-        updating_participant: address,
-        timestamp: makeTimestamp(),
-        fee_schedule: {
-          cap_fees: true,
-          imbalance_penalty: null,
-          proportional: Zero as Int<32>,
-          flat: Zero as Int<32>,
-        },
-      };
-      const msgId = makeMessageId().toString();
+    mergeMap((grouped$) =>
+      combineLatest([grouped$, config$]).pipe(
+        filter(([, { pfsRoom, caps }]) => !!pfsRoom && !!getCap(caps, Capabilities.MEDIATE)),
+        map(([channel, config]) => {
+          const feeSchedule = getFeeScheduleForChannel(channel, config);
+          // using channel feeSchedule above, build a PFSFeeUpdate's schedule payload
+          const schedule: PFSFeeUpdate['fee_schedule'] = {
+            cap_fees: true,
+            flat: feeSchedule.flat,
+            proportional: Zero as Int<32>,
+            imbalance_penalty: null,
+          };
+          return [channel, config, schedule] as const;
+        }),
+        // reactive on channel state and config changes, distinct on schedule's payload
+        distinctUntilChanged(([, , sched1], [, , sched2]) => isEqual(sched1, sched2)),
+        switchMap(([channel, { pfsRoom }, schedule]) => {
+          const message: PFSFeeUpdate = {
+            type: MessageType.PFS_FEE_UPDATE,
+            canonical_identifier: {
+              chain_identifier: BigNumber.from(network.chainId) as UInt<32>,
+              token_network_address: channel.tokenNetwork,
+              channel_identifier: BigNumber.from(channel.id) as UInt<32>,
+            },
+            updating_participant: address,
+            timestamp: makeTimestamp(),
+            fee_schedule: schedule,
+          };
+          const msgId = makeMessageId().toString();
```

Comment:
Yes, I don't like how diff went through this too, but it was needed since this epic's behavior changed: before, it just sent one message for each channel; now it needs to react on channel's and config's changes, and therefore a few checks plus distinction/deduplication conditions had to be nested, which weren't required with the dummy implementation we had before. I'll try to split it a little, but since this is replacing a naive approach with an actual useful one, this is kind of unavoidable.